targets/targets.json

```diff
@@ -3694,6 +3694,42 @@
         "device_name": "STM32F469NI",
         "bootloader_supported": true
     },
+    "SDP-K1": {
```

Comment:
`SDP_K1` should be the name (as any other target). The failure in the build might be related (as it cant find the target) - this could be a bug in our tools or rather unsupported feature 

cc @ARMmbed/mbed-os-tools

Reply:
Would that be a cause of CI test failures? The name with hyphen is consistent across all the supporting files.

Reply:
I believe yes. You should be able to reproduce the build failure locally. To be consistent, it should use `_` instead. (review any other targets). This follows also directory naming `TARGET_` and others.

Reply:
@0xc0170 @bentcooke Fixing the hyphen to underscore, ran the tests on IAR toolchain. The tests were failing to build with"SDP-K1" target. Not sure why that is the case.
Also, for the new changes, should this PR be deleted and a new PR made?